src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineWrappingInBlockCheck.java

```diff
@@ -0,0 +1,440 @@
+ * {@link TokenTypes#LITERAL_FINALLY LITERAL_FINALLY},
+ * {@link TokenTypes#LITERAL_FOR LITERAL_FOR},
+ * {@link TokenTypes#LITERAL_IF LITERAL_IF},
+ * {@link TokenTypes#LITERAL_NEW LITERAL_NEW},
+ * {@link TokenTypes#LITERAL_SWITCH LITERAL_SWITCH},
+ * {@link TokenTypes#LITERAL_SYNCHRONIZED LITERAL_SYNCHRONIZED},
+ * {@link TokenTypes#LITERAL_TRY LITERAL_TRY},
+ * {@link TokenTypes#LITERAL_WHILE LITERAL_WHILE},
+ * {@link TokenTypes#METHOD_DEF METHOD_DEF},
+ * {@link TokenTypes#OBJBLOCK OBJBLOCK},
+ * {@link TokenTypes#SLIST SLIST},
+ * {@link TokenTypes#STATIC_INIT STATIC_INIT}.
+ * </p>
+ *
+ * <p> By default the check will check the following statements:
+ *  {@link TokenTypes#ANNOTATION_DEF ANNOTATION_DEF},
+ *  {@link TokenTypes#CLASS_DEF CLASS_DEF},
+ *  {@link TokenTypes#ENUM_DEF ENUM_DEF},
+ *  {@link TokenTypes#INTERFACE_DEF INTERFACE_DEF}.
+ * </p>
+ *
+ * <p> An example of how to configure the check with default parameters is:
+ * </p>
+ *
+ * <pre>
+ * &lt;module name="EmptyLineWrappingInBlock"/&gt;
+ * </pre>
+ *
+ * <p> Example of declarations with empty line separator at the top of the block
+ * that is expected by the Check by default:
+ * </p>
+ *
+ * <pre>
+ * class Foo {
+ *
+ *     public int method() {
+ *
+ *         return 0;
+ *     }
+ * }
+ * </pre>
+ *
+ * <p> An example how to check wrapping of a {@link TokenTypes#CLASS_DEF CLASS_DEF} and
+ * {@link TokenTypes#INTERFACE_DEF INTERFACE_DEF} only:
+ * </p>
+ *
+ * <pre>
+ * &lt;module name="EmptyLineWrappingInBlock"&gt;
+ *    &lt;property name="tokens" value="CLASS_DEF, INTERFACE_DEF"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ *
+ * <p>
+ * An example how to allow {@link BlockWrapOption#NO_EMPTY_LINE no empty line} after the left curly
+ * brace nor before the right curly brace:
+ * </p>
+ *
+ * <pre>
+ * &lt;module name="EmptyLineWrappingInBlock"&gt;
+ *    &lt;property name="topSeparator" value="no-empty-line"/&gt;
+ *    &lt;property name="bottomSeparator" value="no-empty-line"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ *
+ * <p>
+ * An example how to demand {@link BlockWrapOption#EMPTY_LINE an empty line} after the left curly
+ * brace and before the right curly brace:
+ * </p>
+ *
+ * <pre>
+ * &lt;module name="EmptyLineWrappingInBlock"&gt;
+ *    &lt;property name="topSeparator" value="empty-line"/&gt;
+ *    &lt;property name="bottomSeparator" value="empty-line"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ *
+ * <p>
+ * An example how to demand an {@link BlockWrapOption#EMPTY_LINE empty line} after the left curly
+ * brace, while for the right curly brace an
+ * {@link BlockWrapOption#EMPTY_LINE_ALLOWED empty line is allowed}, but not required:
+ * </p>
+ *
+ * <pre>
+ * &lt;module name="EmptyLineWrappingInBlock"&gt;
+ *    &lt;property name="topSeparator" value="empty-line"/&gt;
+ *    &lt;property name="bottomSeparator" value="empty-line-allowed"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ *
+ * <p> No validation is performed for empty code blocks.
```

Comment:
I do not see a reason of this. Please explain.

Reply:
Consider this config:
topSeparator="empty-line"
bottomSeparator="no-empty-line"
and this case
```
class Foo {

}
```
Should the check report a violation or not? There is an empty line at the bottom which is forbidden.

Reply:
My idea is to have a dedicated option for this case. "emptyBlockOption" or something like that.
Of course, this is make sense only if the top and bottom do not match.

In other hand, config
topSeparator="empty-line"
bottomSeparator="empty-line"
emptyBlockSeparator="no-empty-line"
looks reasonable to me. Example:
```
class BigClass {

  void method() {
    
 }

}

interface Marker {}
```

Reply:
> Should the check report a violation or not? 

no.

>  looks reasonable to me

ok , please put this documentation. One day in future sb will ask us to make it configurable by property.